packages/manager/apps/pci-rancher/src/utils/rancher.ts

```diff
@@ -69,3 +70,80 @@ export const getRancherPlanDescription = (rancherPlan: RancherPlan['name']) => {
       return null;
   }
 };
+
+/**
+ * Extracts drivers and plan information from a switch plan error message.
+ *
+ * @param {string} inputString - The input error message string.
+ * @returns {null|{drivers: string[], plan: string}} - An object containing the drivers and plan, or null if extraction fails.
+ * */
+export function extractDriversAndPlanFromSwitchPlanError(
+  inputString: string,
+): null | { drivers: string[]; plan: string } {
+  try {
+    const openBracketIndex = inputString.indexOf('[');
+    const closeBracketIndex = inputString.indexOf(']');
+    const textBeforeBracket = inputString.substring(0, openBracketIndex).trim();
+    const beginPhrase = 'Unable to switch to plan';
+    if (textBeforeBracket.startsWith(beginPhrase)) {
+      const [plan] =
+        textBeforeBracket.match(/OVHCLOUD_EDITION/) ??
+        textBeforeBracket.match(/STANDARD/);
+      if (!plan) {
+        return null;
+      }
+      const contentInsideBrackets = inputString
+        .substring(openBracketIndex + 1, closeBracketIndex)
+        .trim();
+      if (contentInsideBrackets) {
+        const drivers = contentInsideBrackets.split(',');
+
+        // Create the final array with the text before the bracket and the elements inside the brackets
+        if (drivers.length) {
+          return { drivers, plan };
+        }
+      }
+
+      return null;
+    }
+    return null;
+  } catch (err) {
```

Comment:
What could be the case to catch an error here?
I see no code that can throw an exception in the `try` block